### DISCUSSION TITLE
LED-dimmer zigbee state does not update after usiing psysical button.

### DISCUSSION THREAD

**Discussion by christiaangombert** [Koenkk/zigbee2mqtt#23955] [Devices]
Hi there,
I do have a few LED dimmer switches in my house that I would like to control over zigbee and by physicaly pressing.

The dimmers are of the following brand:
**ION Industries**
model: id200w-zigb
![ION_INDUSTRIES_Zigbee_dimmer_inbouw_0 3_-_200W_iON_Industries_Fase_Afsnijding_LIO00027_m2_big](https://github.com/user-attachments/assets/458d63bc-6b8c-457a-9f88-da32b8238a3e)

The switches can dim a LED lightbulb, and are controlled by both a physical push-rotary button, and via zigbee.
I got the switches to pair with my Zigbee2mqtt network just fine.
In the Zigbee2mqtt user interface, these dimmers are identified as Tuya model TS0601. [https://www.zigbee2mqtt.io/devices/TS0601_dimmer_1_gang_1.html](url)

Taking a closer look at the internal hardware of the dimmer, it seems like the ION Industries dimmer is based on the Tuya ZS3L module (EFR32MG21 based chip).
![IMG_20240913_112749](https://github.com/user-attachments/assets/3d5ae3d4-36c3-4210-b39f-c75fa283338b)

The controls over Zigbee seem to work fine for this dimmer.
Also controling the lights with the physical button works okay.

But the thing that does not seem to work, is an update of the current light state and brightness over Zigbee after using the physical button. Is there any way to get this information feedback of any physical interaction with the button reported to zigbee2mqtt? I would love to hear any suggestions.

Currently I'm running Zigbee2MQTT version 1.39.1 in Proxmox VE.
My coördinator is a first-generation barebone ITead EFR32MG21 Zigbee 3.0 USB Dongle, Model 9888010100045.
The dongle firmware is EZSP v13 version 7.4.3.0.

**Comment by joopmartens** [comment]
Hi Christiaan,

I have the exact same problem and using the same hardware with the only difference that I have upgraded my dongle firmware to 7.4.4
I tried to fix it by creating two reporting rules:

![image](https://github.com/user-attachments/assets/ab1a834a-8429-4abc-a094-3085aa3b098f)

This initially seems to work but at my side it stops working after a short while and can only be fixed by power cycling the dimmers.
I have the same behaviour for all of my 8 ION id200w-zigb dimmers and also looking for a solution.

I have changed the logging of zigbee2mqtt to debug level and when a dimmer stops reporting brightness changes despite the reporting rule it looks like it is not sending any zigbee updates at all at that point in time.

Until now I have not found a solution.

**Comment by joopmartens** [comment]
Another update.

I have had contact with ION and they tell me that the dimmer is always sending the changed dim level via Zigbee when the dim level is changed using the dimmer knob.

They expect it to be a an issue with the gateway/bridge so in this case zigbee2mqtt.

**Reply by christiaangombert** [reply]
That seems to be the case. When i pair the dimmer to a Philips Hue bridge everything works fine. Changes made using the physical button do report instantly in the app too. 
So it seems very likely this issue only happens when using zigbee2mqtt.

**Comment by EdeMeijer** [comment]
Today I've run into the same or a similar issue. I created the same reporting rules as @joopmartens and after doing so, I'm in fact receiving state updates after turning the light on/off using the physical switch. However, regardless of the actual state, the dimmer always reports the last state value that it _received_ from zigbee2mqtt, not the actual state.

This seems quite strange to me. I sent an email to the manufacturer to ask for clarifications and help, I'll let you know if I learn more.

**Comment by christiaangombert** [comment]
Yesterday I tried to get the status updates of the ION working with a new Zigbee coordinator, the SMlight slzb-06m (EFR32MG21 based). It had the firmware revision: 20241127, SDK: 7.4.5, adapter: Ember. I used a new Zigbee2mqtt installation (in proxmox) for this too. Still the problem occurred as described in the OP. I got no status updates when operating the dimmer using the physicals button.

After that I did use ZHA with the same SMlight slzb-06m coordinator. With ZHA the status report of the ION dimmers worked flawlessly. My guess is that the ION does communicate the status to the coordinator just fine. It's Zigbee2mqtt that is not picking up the messages.

**Reply by EdeMeijer** [reply]
Alright. I received a prompt but slightly rude reply from ION INDUSTRIES that their device works fine and they can't fix the gateway software. At least it's good to know that the device is in fact not to blame, but that the zigbee2mqtt integration is lacking.

I'm still still pretty new to all this stuff, but I'm interested in learning about how these integrations work exactly and willing to look into this further at some point in the near-ish future. 

Edit: by the way, I'm using a SLZB-06, not the M version, and just upgraded all the firmware.

**Comment by Nikneem0** [comment]
I also struggled with this for a while. Working well in ZHA but not with Zigbee2MQTT. But I found the solution!
**Connect the Neutral wire!!!**
It works for my three dimmers.

**Reply by christiaangombert** [reply]
I'll try wiring a neutral wire somewhere in the coming weeks. I'll let you guys know if the fix also works for me.

**Reply by joopmartens** [reply]
I also have this problem and all my dimmers are already connected using the neutral wire from the start.
So unfortunately it does not seems to be a solution for me.

I also have to point out that after a cold boot of the dimmers after a powercut, the brightness level is already reported when it is adjusted using the knob via Zigbee but this stops functioning after a while while the on/off state keeps working.

I'm curious if this will be different in your case.

**Reply by TimGW** [reply]
I also have the same issues with both the regular single unit ( 90.500.020 ) as with the master/slave set (90.500.090). Neutral wires were connected prior to the Z2M integration. Eventually I gave up and migrated to ZHA where everything seem to be working as it should for these ION dimmers.

**Reply by Nikneem0** [reply]
I had one dimmer already connected with a Neutral wire (because the load is too low for Life wire only). I noticed that it was reporting physical actions. That’s why I got the idea of connecting Neutral also to the other two. 
Over the past month I have noticed (on all my 3 dimmers) that physically switching the dimmer on or off results in a proper status update but changing the brightness with the physical knob almost always goes unnoticed by HA. So, there still is a problem!

**Reply by duncanhancox** [reply]
the same issue happens with the similar candio and samotech push / rotary zigbee dimmers with neutral connected - i gave up using them in the end

**Comment by EdeMeijer** [comment]
It's been 1,5 years since my last attempt, but Claude helped me finally fix the issue for my particular model. Check out https://github.com/Koenkk/zigbee-herdsman-converters/pull/12235, hopefully it can still help some of you.

**Comment by EdeMeijer** [comment]
@christiaangombert my fix landed in a main zigbee2mqtt release, and at least for me, the ion dimmers give feedback to HA properly in a fresh installation.

**Comment by wimensing** [comment]
Since the zigbee2mqtt release where _TZE200_ykgar0ow is now supported, I have something else:

When I turn it on, the status updates correctly in HA.
When I turn it off, the status updates correctly in HA.
But then when the hourly status updates comes along, the dimmer reports ON.

The dimmer is not installed with a N wire. 
I have a also installed a dimmer bypass.
Can this bypas be the cause of the dimmer reporting ON?

**Reply by EdeMeijer** [reply]
Now that you mention it, I might be seeing the same thing. Back to the drawing board, I guess

**Reply by EdeMeijer** [reply]
@wimensing found the issue, https://github.com/Koenkk/zigbee-herdsman-converters/pull/12431 fixes it. In the meantime, you can manually fix it in the Z2M UI by setting the max reporting values to 0 for the dimmer bindings (on/off and level).

**Comment by wimensing** [comment]
@EdeMeijer Great! And thanks for the pointer to the manual fix.
